src/dmd/toobj.d

```diff
@@ -503,12 +512,13 @@ void toObjFile(Dsymbol ds, bool multiobj)
                         Dsymbol sm = (*pc.members)[i];
                         //printf("sm = %s %s\n", sm.kind(), sm.toChars());
                         if (sm.hasPointers())
-                            goto L2;
+                        {
+                            flags &= ~ClassFlags.noPointers;  // not no-how, not no-way
+                            break Louter;
+                        }
                     }
                 }
             }
-            flags |= ClassFlags.noPointers;
-          L2:
```

Comment:
Please avoid seemingly unrelated refactorings in bugfixes or mention why they were necessary.